modules/src/ics07_tendermint/amino_types.rs

```diff
@@ -180,24 +179,45 @@ pub struct Version {
, 0x4, 0x4a, 0x20, 0x99, 0x5a, 0xbf, 0x39, 0x15, 0x1d, 0xea, 0xc4, 0xcc, 0xbc, 0x7c, 0x30, 0xf3, 0xa4, 0xeb, 0x52, 0x23, 0x5d, 0x58, 0x9a, 0xab, 0x57, 0x3a, 0x96, 0x6b, 0x7d, 0xa3, 0xb0, 0x6c, 0x82, 0x94, 0x4, 0x52, 0x20, 0xe5, 0xe5, 0x66, 0xc4, 0x1e, 0xd5, 0x7e, 0x3f, 0xf8, 0xcc, 0x10, 0xf1, 0x84, 0x17, 0x87, 0x88, 0xb8, 0xfa, 0xa6, 0x2, 0xb0, 0x7c, 0xf1, 0xf4, 0x25, 0x21, 0x7b, 0xd8, 0x17, 0x9f, 0x1f, 0x24, 0x5a, 0x20, 0x41, 0xca, 0xfa, 0xe3, 0x1c, 0xc7, 0xf, 0x58, 0x1, 0xfa, 0x10, 0x16, 0xa2, 0xdd, 0x54, 0xa9, 0xbc, 0xb8, 0x20, 0x1b, 0x5b, 0x38, 0x99, 0x19, 0xfe, 0x99, 0x76, 0x76, 0x25, 0x32, 0xc5, 0x16, 0x62, 0x20, 0x9, 0x2e, 0x5, 0x86, 0x30, 0x24, 0x7e, 0xd6, 0x0, 0x98, 0x63, 0xa1, 0x2e, 0xee, 0x11, 0x7d, 0x26, 0xcd, 0x9d, 0x8, 0xb5, 0xad, 0xca, 0xab, 0x37, 0xf2, 0xab, 0x35, 0xdb, 0x47, 0x5a, 0x37, 0x6a, 0x20, 0x73, 0x86, 0x5d, 0xb0, 0x8f, 0x49, 0xd5, 0x84, 0x28, 0x90, 0x5d, 0x38, 0x9a, 0xb4, 0xca, 0x4b, 0x96, 0xe4, 0x5a, 0x32, 0x6, 0xc7, 0xa6, 0x9d, 0x43, 0xa5, 0xdc, 0x73, 0x72, 0xe6, 0x7, 0x14, 0x72, 0x14, 0xf0, 0xa6, 0xcb, 0xba, 0x15, 0x16, 0x45, 0x22, 0xa5, 0xcc, 0xf4, 0x56, 0x71, 0x3f, 0xb2, 0xfe, 0x63, 0xc5, 0xaa, 0x67, 0x12, 0xb2, 0x1, 0x8, 0x4, 0x10, 0x1, 0x1a, 0x48, 0xa, 0x20, 0x9f, 0x14, 0xcb, 0x6b, 0x8c, 0x12, 0xd7, 0x6c, 0xee, 0x98, 0xec, 0x21, 0xee, 0x27, 0xd3, 0xb2, 0xa2, 0x6d, 0xc, 0xde, 0x93, 0x3c, 0xd6, 0xf7, 0x15, 0x83, 0xef, 0xaa, 0x58, 0x27, 0xea, 0xae, 0x12, 0x24, 0x8, 0x3, 0x12, 0x20, 0x87, 0x8, 0xa, 0x39, 0xcf, 0xe3, 0x36, 0xa6, 0x63, 0xea, 0x5c, 0xd, 0x9b, 0xde, 0xef, 0x49, 0x3a, 0xbd, 0xc1, 0x8b, 0x7c, 0x3e, 0x3e, 0xc, 0x8e, 0x66, 0x13, 0x54, 0xc1, 0x83, 0x14, 0x32, 0x22, 0x62, 0x8, 0x2, 0x12, 0x14, 0xf0, 0xa6, 0xcb, 0xba, 0x15, 0x16, 0x45, 0x22, 0xa5, 0xcc, 0xf4, 0x56, 0x71, 0x3f, 0xb2, 0xfe, 0x63, 0xc5, 0xaa, 0x67, 0x1a, 0x6, 0x8, 0x80, 0xe5, 0xb4, 0xf0, 0x5, 0x22, 0x40, 0xa3, 0xd8, 0x2b, 0xc2, 0xf7, 0x84, 0x5d, 0xbc, 0x92, 0x4d, 0xbf, 0x70, 0x9a, 0xa6, 0xd, 0x2d, 0x68, 0x24, 0x93, 0x86, 0xbb, 0x5b, 0xd6, 0x7, 0x57, 0x11, 0x13, 0xc5, 0x36, 0x45, 0x34, 0x31, 0xe, 0x3a, 0xd2, 0xa4, 0xc2, 0x1f, 0x32, 0xc7, 0x6, 0x78, 0xae, 0x66, 0x8c, 0xa, 0x8a, 0x5f, 0x75, 0xb3, 0xa2, 0x34, 0x68, 0xaa, 0xfc, 0xe1, 0xe5, 0xa7, 0xa8, 0xd5, 0xf9, 0x28, 0xe6, 0x2, 0x12, 0x82, 0x1, 0xa, 0x3f, 0xa, 0x14, 0xf0, 0xa6, 0xcb, 0xba, 0x15, 0x16, 0x45, 0x22, 0xa5, 0xcc, 0xf4, 0x56, 0x71, 0x3f, 0xb2, 0xfe, 0x63, 0xc5, 0xaa, 0x67, 0x12, 0x25, 0x16, 0x24, 0xde, 0x64, 0x20, 0x34, 0x97, 0xd, 0xb3, 0x8d, 0x45, 0x6b, 0x77, 0xf2, 0x10, 0x85, 0x96, 0x3f, 0x43, 0x80, 0x86, 0x4c, 0x90, 0xcc, 0xe9, 0x91, 0x25, 0xc6, 0x7, 0xa7, 0xcc, 0xfe, 0xfb, 0x39, 0x25, 0xfc, 0xff, 0x18, 0xa, 0x12, 0x3f, 0xa, 0x14, 0xf0, 0xa6, 0xcb, 0xba, 0x15, 0x16, 0x45, 0x22, 0xa5, 0xcc, 0xf4, 0x56, 0x71, 0x3f, 0xb2, 0xfe, 0x63, 0xc5, 0xaa, 0x67, 0x12, 0x25, 0x16, 0x24, 0xde, 0x64, 0x20, 0x34, 0x97, 0xd, 0xb3, 0x8d, 0x45, 0x6b, 0x77, 0xf2, 0x10, 0x85, 0x96, 0x3f, 0x43, 0x80, 0x86, 0x4c, 0x90, 0xcc, 0xe9, 0x91, 0x25, 0xc6, 0x7, 0xa7, 0xcc, 0xfe, 0xfb, 0x39, 0x25, 0xfc, 0xff, 0x18, 0xa],
```

Comment:
I plugged in this code into `client_state_test.go`to generate these test-vectors:
```go
func (suite *TendermintTestSuite) TestGenerateEncodingVectors() {
	testCases := []struct {
		name        string
		clientState ibctmtypes.ClientState
	}{
		{
			name:        "ID only",
			clientState: ibctmtypes.ClientState{ID: chainID},
		},
		{
			name:        "ID, trust period",
			clientState: ibctmtypes.ClientState{ID: chainID, TrustingPeriod: trustingPeriod},
		},
		{
			name:        "ID, trust, unbonding period",
			clientState: ibctmtypes.ClientState{ID: chainID, TrustingPeriod: trustingPeriod, UnbondingPeriod: ubdPeriod},
		},
		{
			name:        "with full header",
			clientState: ibctmtypes.NewClientState(chainID, trustingPeriod, ubdPeriod, suite.header),
		},
	}

	for _, tc := range testCases {
		tc := tc
		fmt.Println("---------------------------")
		fmt.Println(tc.name)
		b, _ := suite.cdc.MarshalBinaryLengthPrefixed(tc.clientState)
		fmt.Println(len(b))	
		fmt.Println("---------------------------")
	}
}
```